src/metric-engine/src/engine.rs

```diff
@@ -121,23 +122,39 @@ impl RegionEngine for MetricEngine {
         &self,
         region_id: RegionId,
         request: RegionRequest,
-    ) -> Result<AffectedRows, BoxedError> {
+    ) -> Result<RegionHandleResult, BoxedError> {
+        let mut extension_return_value = HashMap::new();
```

Comment:
One unresolved problem is how to merge multiple extensions in batch request. I'm not sure which one is better, making a new batching interface or implement a `merge()` method